services/utils.py

```
@@ -94,22 +153,47 @@ def load_json(self, filename: str, date: Optional[datetime] = None) -> Optional[
         
-        # Move the file
-        temp_path.rename(persistent_path)
+        # Check if destination exists and overwrite is not allowed
+        if persistent_path.exists() and not overwrite:
+            raise FileExistsError(f"Destination file already exists: {persistent_path}. Set overwrite=True to allow replacement.")
+        
+        # Perform atomic replace using os.replace
+        try:
+            os.replace(temp_path, persistent_path)
+        except (OSError, PermissionError) as e:
+            raise RuntimeError(f"Failed to persist file from {temp_path} to {persistent_path}: {e}") from e
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)